src/formio/components/Children/Children.jsx

```diff
@@ -0,0 +1,238 @@
+      icon: 'children',
+      group: 'custom_special',
+      weight: 300,
+      schema: Children.schema(),
+    };
+  }
+
+  get defaultSchema() {
+    return Children.schema();
+  }
+
+  get emptyValue() {
+    return [];
+  }
+
+  getValue() {
+    return this.dataValue;
+  }
+
+  setValue(value, flags = {}) {
+    if (!value) {
+      value = [];
+    }
+
+    // Normalize each child to ensure `__selected` and `__addedManually` are always a
+    // boolean (used to track the added/selected children by the user)
+    value = value.map(child => ({
+      ...child,
+      __selected: typeof child.__selected === 'boolean' ? child.__selected : false,
+      __addedManually: typeof child.__addedManually === 'boolean' ? child.__addedManually : false,
+    }));
+
+    const changed = super.setValue(value, flags);
+
+    // Re-render if the value was actually changed
+    if (changed) {
+      this.renderReact();
+    }
+
+    return changed;
+  }
+
+  isEmpty() {
+    return !this.dataValue || this.dataValue?.length === 0;
+  }
+
+  destroy() {
+    const container = this.refs.childrenContainer;
+    if (!this.component?.hidden && container) this.reactRoot.unmount();
+    super.destroy();
+  }
+
+  onFormikChange(value) {
+    this.updateValue(value, {modified: true});
+
+    // `enableValidationPlugins` forces the component to be validateOn = 'blur', which
+    // surpresses the validators due to onChange events.
+    // Since this is a composite event, we need to fire the blur event ourselves and
+    // schedule the validation to run.
+    // Code inspired on Formio.js' `src/components/_classes/input/Input.js`, in
+    // particular the `addFocusBlurEvents` method.
+    //
+    if (this.component.validateOn === 'blur') {
+      if (this._debouncedBlur) {
+        this._debouncedBlur.cancel();
+      }
+
+      this._debouncedBlur = debounce(() => {
+        this.root.triggerChange(
+          {fromBlur: true},
+          {
+            instance: this,
+            component: this.component,
+            value: this.dataValue,
+            flags: {fromBlur: true},
+          }
+        );
+      }, 50);
+
+      this._debouncedBlur();
+    }
+  }
+
+  render() {
+    return super.render(
+      `<div ref="element">
+          ${this.renderTemplate('children', this.renderContext)}
+        </div>`
+    );
+  }
+
+  openChildModal(child = null) {
+    this.state.isChildModalOpen = true;
+    this.state.childBeingEdited = child;
+    this.renderReact();
+  }
+
+  closeChildModal() {
+    this.state.isChildModalOpen = false;
+    this.state.childBeingEdited = null;
+    this.renderReact();
+  }
+
+  handleChildSave(newChild) {
+    const childExists = newChild.__addedManually === true;
+    const oldChild = this.state.childBeingEdited;
+
+    let updated;
+    if (childExists) {
+      // Update the existing child
+      updated = this.dataValue.map(child =>
+        child.bsn === oldChild.bsn ? {...child, ...newChild} : child
+      );
```

Comment:
@robinmolen apart from your comments I changed this here too because each time the user would modify the BSN a new child was added (which didn't make sense to me, if I want to modify the specific child I want to do exactly this).

Reply:
Thanks for the proper review/testing!Yes, my first thought was to disable the BSN field in the edit mode but couldn't do that (requires changes in the form renderer I guess). I think I have found somthing though